src/controller/CHIPDevice.h

```diff
@@ -173,6 +174,7 @@ class DLL_EXPORT Device : public Messaging::ExchangeDelegate, public SessionEsta
         mAdminId         = admin;
         mStorageDelegate = params.storageDelegate;
         mCredentials     = params.credentials;
+        mTrustedRootId   = params.trustedRoot;
```

Comment:
`param.trustedRoot` is not being populated in the code. We are assigning it (nullptr) to `mTrustedRootId`, and dereferencing `mTrustedRootId` later in `EstablishCASESession` without any checks.

Who is supposed to fill in this value, and how would they determine what to fill?
Also, we should test that this is not null before dereferencing it.

Reply:
In our implementation of the Commissioning Demo, we use CHIPDeviceController to trigger a CASESession (instead of the CHIPDevice). From CHIPDeviceController we can retrieve the Trusted Root ID from the mRootKeyId member variable. We think it's a better alternative to deal with this. The code we added for CHIPDevice is there to avoid build failure. If you agree, we could drop the EstablishCASESession from CHIPDevice and move it to CHIPDeviceController in a later commit.